src/ray/core_worker/future_resolver.h

```diff
@@ -0,0 +1,68 @@
+#ifndef RAY_CORE_WORKER_FUTURE_RESOLVER_H
+#define RAY_CORE_WORKER_FUTURE_RESOLVER_H
+
+#include <memory>
+
+#include "ray/common/id.h"
+#include "ray/core_worker/store_provider/memory_store/memory_store.h"
+#include "ray/protobuf/core_worker.pb.h"
+#include "ray/rpc/worker/core_worker_client.h"
+
+namespace ray {
+
+/// Max time between requests to the owner to check whether the object has been
+/// evicted.
+const int kWaitObjectEvictionMilliseconds = 100;
```

Comment:
This seems quite low for non-test setting, why not 5 seconds at least? Performance on the failure path does not really matter that much.

Reply:
Discussed offline, but for posterity, we set the timeout low here because the submission of tasks that depend on these objects have to wait for the futures to get resolved. Later on, we can change this to queue the callbacks at the owner and have owner push a reply when the future is resolved (will add a TODO for this).